### PR TITLE
Add grid-extensions.md

### PR DESCRIPTION
**Pull request by brminnick**
This PR adds `grid-extensions.md` which defines the APIs in `GridRowsColumns.cs` and `GridRowsColumns.cs`

### CODE REVIEW

**Comment by PRMerger10** [comment]
Microsoft authors of articles need to make updates through the private repository, so the updates can be staged and validated by current validation rules. After you review staging and fix all validation issues, you can add the sign-off comment to let the PR reviewers know the updates are ready to be merged. 

It's easy to access the private repo version of your articles in GitHub. Just add *-pr* to the URL, as follows: 

 &nbsp;Public repo URL: `https://github.com/MicrosoftDocs/{docs-repo}` 
 &nbsp;Private repo URL: `https://github.com/MicrosoftDocs/{docs-repo-pr}` 

 We have closed this pull request. Please resubmit the updates through the private repo. For more information, see the [contributor guide](https://docs.microsoft.com/contribute/).

**Comment by PRMerger12** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 File Change Percent | This PR contains file(s) with more than 20% file change.

**Comment by PRMerger12** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 File Change Percent | This PR contains file(s) with more than 20% file change.

**Comment by PRMerger18** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 File Change Percent | This PR contains file(s) with more than 20% file change.

**Comment by PRMerger14** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 File Change Percent | This PR contains file(s) with more than 20% file change.

**Comment by PRMerger16** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 File Change Percent | This PR contains file(s) with more than 20% file change.

**Comment by PRMerger9** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 File Change Percent | This PR contains file(s) with more than 20% file change.

**Comment by PRMerger8** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 File Change Percent | This PR contains file(s) with more than 20% file change.

File: docs/maui/markup/extensions/grid-extensions.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+The `RowSpan` method allows us to define how many Grid Rows a control will span across. I.e. If our `Grid` has 3 Rows, `.RowSpan(3)` will ensure the control spans across all 3 Columns.
+
+Here's an example of a `Button` that spans vertically across 3 Rows:
+```cs
+new Button()
+    .Text("This Button Spans Across 3 Grid Rows")
+    .RowSpan(3)
+```
+
+### All&lt;TEnum>
```
Comment (bijington) [inline docs/maui/markup/extensions/grid-extensions.md:227]: Does this need an `&gt;` on the right?

File: docs/maui/markup/extensions/grid-extensions.md [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+The `ColumnSpan` method allows us to define how many Grid Columns a control will span across. I.e. If our `Grid` has 3 Columns, `.ColumnSpan(3)` will ensure the control spans across all 3 Columns.
+
+Here's an example of a `Button` that spans horizontally across 3 Columns:
+```cs
+new Button()
+    .Text("This Button Spans Across 3 Grid Columns")
+    .ColumnSpan(3)
+```
+
+### All&lt;TEnum>
```
Comment (bijington) [inline docs/maui/markup/extensions/grid-extensions.md:251]: Does this need an `&gt;` on the right?

File: docs/maui/markup/extensions/grid-extensions.md [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+```cs
+enum Column { Description, UserInput }
+
+// ...
+new Button()
+    .Text("This Button Spans Vertically Across Every Row Defined in our Enum")
+    .ColumnSpan(All<Column>());
+```
+
+## Last&lt;TEnum>
```
Comment (bijington) [inline docs/maui/markup/extensions/grid-extensions.md:264]: Does this need an `&gt;` on the right?

**Review by bijington — APPROVED** [review]
@brminnick thanks for this! Only 3 minor points that I wasn't sure on but all related.

**Comment by PRMerger6** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 File Change Percent | This PR contains file(s) with more than 20% file change.

**Comment by brminnick** [comment]
Thanks Shaun! Oddly enough, we only need to use `&lt;` and don't need to use `&gt;` 🤷‍♂️ 

image

**Pull request merged**